src/Compilers/VisualBasic/Portable/BoundTree/Expression.vb

```diff
@@ -1167,7 +1167,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
         Private ReadOnly Property IHasArgumentsExpression_ArgumentsInParameterOrder As ImmutableArray(Of IArgument) Implements IHasArgumentsExpression.ArgumentsInParameterOrder
             Get
-                Return BoundCall.DeriveArguments(Me.Arguments, Me.ConstructorOpt.Parameters)
+                Return If(Me.ConstructorOpt Is Nothing, ImmutableArray(Of IArgument).Empty, BoundCall.DeriveArguments(Me.Arguments, Me.ConstructorOpt.Parameters))
```

Comment:
I don't know of a case where a valid program could produce a null constructor, but @AlekseyTs will have more of the language subtleties to mind.

Someone (meaning myself or @genlu ) needs to make a systematic pass through the IOperation implementations and protect against null properties or, more generally, malformed bound trees due to invalid source.